### DISCUSSION TITLE
Oracle and raw uuid identifiers

### DISCUSSION THREAD

**Discussion by smarley** [doctrine/doctrine2#9389] [Support Questions]
Hi everyone!

TL;DR: why i can't use `convertToDatabaseValueSQL` with identifiers? Is there any workarounds?

I have a project with doctrine + oracle + uuid identifiers. For handling uuid used custom type, and uuid stored as RAW type.

All queries produced by `$em->flush()` after updating entities fields or `$repository->find($id)` is like
```
WHERE id = :param
```
and this produces full table scan, ignoring index (fair enough):

```
UPDATE users SET name = :param1 WHERE id = :param2
 
Plan hash value: 3217041765
 
----------------------------------------------------------------------------
| Id  | Operation          | Name  | Rows  | Bytes | Cost (%CPU)| Time     |
----------------------------------------------------------------------------
|   0 | UPDATE STATEMENT   |       |       |       |     3 (100)|          |
|   1 |  UPDATE            | USERS |       |       |            |          |
|*  2 |   TABLE ACCESS FULL| USERS |     1 |    31 |     3   (0)| 00:00:01 |
----------------------------------------------------------------------------
 
Predicate Information (identified by operation id):
---------------------------------------------------
 
   2 - filter(RAWTOHEX("ID")=:PARAM2)
```

As i see, Doctrine can't use `convertToDatabaseValueSQL` for identifiers. I would really appreciate some advice on how to deal with this situation.

With

```php
public function convertToDatabaseValueSQL($sqlExpr, AbstractPlatform $platform)
{
    return sprintf('HEXTORAW(%s)', $sqlExpr);
}
```
and QueryBuilder query like this

```php
$this->createQueryBuilder('p')
            ->delete()
            ->where('p.paramId = :paramId')
            ->setParameter('paramId', $paramId, UuidHexType::NAME)
            ->getQuery()
            ->execute();
```
everything work just fine,  produces `WHERE id = HEXTORAW(:param2)` with INDEX RANGE SCAN
